controllers/apps/apimanager_status_reconciler.go

```diff
@@ -0,0 +1,234 @@
+	for _, dcName := range expectedDeploymentNames {
+		existingDeploymentConfig := &appsv1.DeploymentConfig{}
+		err := s.Client().Get(context.Background(), types.NamespacedName{Namespace: s.apimanagerResource.Namespace, Name: dcName}, existingDeploymentConfig)
+		if err != nil && !errors.IsNotFound(err) {
+			return nil, err
+		}
+		if err != nil && errors.IsNotFound(err) {
+			continue
+		}
+
+		for _, ownerRef := range existingDeploymentConfig.GetOwnerReferences() {
+			if ownerRef.UID == s.apimanagerResource.UID {
+				dcs = append(dcs, *existingDeploymentConfig)
+				break
+			}
+		}
+	}
+	sort.Slice(dcs, func(i, j int) bool { return dcs[i].Name < dcs[j].Name })
+
+	return dcs, nil
+}
+
+func (s *APIManagerStatusReconciler) apimanagerAvailableCondition(existingDeployments []appsv1.DeploymentConfig) (common.Condition, error) {
+	deploymentsAvailable := s.deploymentsAvailable(existingDeployments)
+
+	defaultRoutesReady, err := s.defaultRoutesReady()
+	if err != nil {
+		return common.Condition{}, err
+	}
+
+	newAvailableCondition := common.Condition{
+		Type:   appsv1alpha1.APIManagerAvailableConditionType,
+		Status: v1.ConditionFalse,
+	}
+
+	if deploymentsAvailable && defaultRoutesReady {
+		newAvailableCondition.Status = v1.ConditionTrue
+	}
+
+	return newAvailableCondition, nil
+}
+
+func (s *APIManagerStatusReconciler) defaultRoutesReady() (bool, error) {
+	wildcardDomain := s.apimanagerResource.Spec.WildcardDomain
+	routeHosts := []string{
+		fmt.Sprintf("backend-%s.%s", *s.apimanagerResource.Spec.TenantName, wildcardDomain),                // Backend Listener route
+		fmt.Sprintf("api-%s-apicast-production.%s", *s.apimanagerResource.Spec.TenantName, wildcardDomain), // Apicast Production default tenant Route
+		fmt.Sprintf("api-%s-apicast-staging.%s", *s.apimanagerResource.Spec.TenantName, wildcardDomain),    // Apicast Staging default tenant Route
+		fmt.Sprintf("master.%s", wildcardDomain),                                                           // System's Master Portal Route
+		fmt.Sprintf("%s.%s", *s.apimanagerResource.Spec.TenantName, wildcardDomain),                        // System's default tenant Developer Portal Route
+		fmt.Sprintf("%s-admin.%s", *s.apimanagerResource.Spec.TenantName, wildcardDomain),                  // System's default tenant Admin Portal Route
+	}
+
+	listOps := []client.ListOption{
+		client.InNamespace(s.apimanagerResource.Namespace),
+	}
+
+	routeList := &routev1.RouteList{}
+	err := s.Client().List(context.TODO(), routeList, listOps...)
+	if err != nil {
+		return false, fmt.Errorf("Failed to list routes: %w", err)
+	}
+
+	var routes []routev1.Route
+	for _, route := range routeList.Items {
+		routes = append(routes, route)
+	}
+	sort.Slice(routes, func(i, j int) bool { return routes[i].Name < routes[j].Name })
+
+	allDefaultRoutesReady := true
+	for _, routeHost := range routeHosts {
+		routeFound := false
+		var matchedRoute *routev1.Route
+		for _, route := range routes {
```

Comment:
minor proposal:

add to ` pkg/helper/route.go` RouteFindByHost method:

```
// RouteFindByHost returns the smallest index i at which a route with a given host is found
// or -1 if there is no such index.                                 
func RouteFindByHost(a []routev1.Route, routeHost string) int {                      
    for i, n := range a {                                           
        if n.Spec.Host == routeHost {                                            
            return i                                                
        }                                                           
    }                                                               
    return -1                                                       
}                                                                   

```

Reply:
:+1:  changed